src/Sarif.Converters/ContrastSecurityConverter.cs

```diff
@@ -0,0 +1,1364 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Xml;
+using System.Xml.Schema;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    /// <summary>
+    /// Converts exported Contrast Security XML report files to sarif format
+    /// </summary>
+    ///<remarks>
+    ///</remarks>
+    internal sealed class ContrastSecurityConverter : ToolFileConverterBase
+    {
+        private const string ContrastSecurityRulesData = "Microsoft.CodeAnalysis.Sarif.Converters.RulesData.ContrastSecurity.sarif";
+
+        private IDictionary<string, Rule> _rules;
+        private HashSet<FileData> _files;
+
+        /// <summary>
+        /// Convert Contrast Security log to SARIF format stream
+        /// </summary>
+        /// <param name="input">Contrast log stream</param>
+        /// <param name="output">output stream</param>
+        /// <param name="dataToInsert">Optionally emitted properties that should be written to log.</param>
+        public override void Convert(Stream input, IResultLogWriter output, OptionallyEmittedData dataToInsert)
+        {
+            if (input == null)
+            {
+                throw (new ArgumentNullException(nameof(input)));
+            }
+
+            if (output == null)
+            {
+                throw (new ArgumentNullException(nameof(output)));
+            }
+
+            LogicalLocations.Clear();
+
+            _files = new HashSet<FileData>(FileData.ValueComparer);
+
+            var context = new ContrastLogReader.Context();
+
+            // 1. First we initialize our global rules data from the SARIF we have embedded as a resource
+            Assembly assembly = typeof(ContrastSecurityConverter).Assembly;
+            SarifLog sarifLog;
+
+            using (var stream = assembly.GetManifestResourceStream(ContrastSecurityConverter.ContrastSecurityRulesData))
+            using (var streamReader = new StreamReader(stream))
+            {
+                string prereleaseRuleDataLogText = streamReader.ReadToEnd();
+                PrereleaseCompatibilityTransformer.UpdateToCurrentVersion(prereleaseRuleDataLogText, forceUpdate: true, Newtonsoft.Json.Formatting.Indented, out string currentRuleDataLogText);
```

Comment:
>UpdateToCurrentVersion [](start = 51, length = 22)

The Contrast Security rules log is in 10-10 format -- containing a rules dictionary -- and it was easier to do this than to update the file. In a separate PR we can run Multitool to update it to 2019-01-09.